### ISSUE TITLE
GUI: Foldable group boxes

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#325]
Allow groups of UI elements to be hidden with a single click, to de-clutter toolbars.

**Comment by jdtournier** [comment]
Another option is to have scrollable tools, so that the window can be smaller than the contents. I'd tried this very early on, but it proved a little finicky to get it to behave right. Specifically, I don't necessarily want tools to be scrollable horizontally, and I'd like them to be full-sized when shown if possible - i.e. only shrink the window vertically relative to its contents if it won't fit on the screen. I reckon it's worth having another look into this, in addition to the foldable UI elements (or maybe instead of them, it sounds like it would be much simpler to implement...).

**Comment by Lestropie** [comment]
I think both are worth looking into. Even if vertical scrolling tools are there, there might be times when you just don't want a huge amount of UI elements complicating your screen when there are entire sections that you're not going to touch.

**Comment by jdtournier** [comment]
I agree - especially with some of the larger tools that have been produced recently. That said, I do think the scrollable version might be the easier of the two...
